### ISSUE TITLE
Invert image colors?

### ISSUE THREAD

**Issue by goodloetb2** [stevencohn/OneMore#1258]
My apologies if this feature already exists. I was going through the description and issue log but couldn't find any mention of whether or not OneMore can invert image color. I work in dark mode and my professor likes to work from PDFs of powerpoint slide decks. It would be great if, when I insert printouts of those slides if I could then invert their colors so I have pages with black backgrounds instead of white backgrounds.

**Comment by stevencohn** [comment]
Hi,

There is no equivalent command for this. I could add an “invert colors” option, but that would only really work OK for black and white images, and it would affect the entire image, so anything “gray” might look really strange, like a photo negative.

I’m afraid I don’t have the image processing background needed to do this cleanly.

Best Regards,
Steven



From: goodloetb2 ***@***.***>
Sent: Sunday, January 14, 2024 2:00 PM
To: stevencohn/OneMore ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [stevencohn/OneMore] Invert image colors? (Issue #1258)


My apologies if this feature already exists. I was going through the description and issue log but couldn't find any mention of whether or not OneMore can invert image color. I work in dark mode and my professor likes to work from PDFs of powerpoint slide decks. It would be great if, when I insert printouts of those slides if I could then invert their colors so I have pages with black backgrounds instead of white backgrounds.

**Comment by PaulJamesTaylor** [comment]
@goodloetb2 I have the same problem when I'm working at night in a dark room.  I have PDF docs I take notes on which are glaring white and turning on dark mode in Onenote doesn't work to invert the white color background.  Instead I invert my whole computer.  I don't recall what the option is called but I did some research in the past on it and found that I could enable it with the key combo (WindowsKey+ctrl+c) - this is a built in Windows feature.  This inverts all applications and some of what is on the screen does look like a photo negative which is annoying (for example Outlook looks terrible), but if I'm only using Onenote it works fine.  I also sometimes use the f.lux and keep Onenote in light mode (a different approach obviously).

**Comment by goodloetb2** [comment]
I would submit that I don't think the photo negative thing would be an issue, although maybe it would be worth disclosing 🤷 For context, I have a manual way of doing this which is that I use SnagIt with a filter effect which I've then tied to a keyboard shortcut. When I hit the shortcut, I get prompted to select the area of the screen I want to capture. After I make the selection, Snagit inverts the colors then puts the result in my clipboard where I can then paste it to OneNote. If the screencap has an image in it, it definitely looks like a negative. But usually I'm doing this in a Zoom call where someone is presenting a slide deck... the important part is that the otherwise white background is now black and I have the words, which are now white. As you can imagine, however, the usefulness of this method falls off considerably if what I have is an 80 page deck.

**Comment by stevencohn** [comment]
I'll play and let you know...

**Comment by stevencohn** [comment]
Initial testing is good. Might need to work on clarity...

Example PDF import

![image](https://github.com/stevencohn/OneMore/assets/9964757/8082e0fc-98e3-488e-a525-6c47bcbf551e)

Inverted

![image](https://github.com/stevencohn/OneMore/assets/9964757/de86abf2-e1f0-4ac7-8f08-0f3dc0c46675)

**Comment by goodloetb2** [comment]
> Initial testing is good. Might need to work on clarity...
> 

This looks good!!!

**Comment by stevencohn** [comment]
Images can get funky

<img width="200" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/4bc30171-f2a2-4ba6-b49d-ccdf6f6b9dc6">

**Comment by goodloetb2** [comment]
> Images can get funky
> 
Totally... but you can seamlessly write across the background and the page 🥰

**Comment by PaulJamesTaylor** [comment]
@stevencohn  The PDF docs I work on are all technical documents with no photo images.  I would find this useful (although machine level inversion does the same thing).

**Comment by stevencohn** [comment]
I learned something new today. Apparently, this was enabled in Win10 but must be enabled in Win11. It called Accessibility / Color Filters. Nice.

**Comment by stevencohn** [comment]
Added an Invert style to the Resize and Adjust dialog.

Also added a new Stylize Images command:

<img width="250" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/3d67b80d-af47-42ab-b87b-ef5ef5456447">

**Comment by goodloetb2** [comment]
Seeing as you weren't able to replicate my other issue, I can't help but wonder what's going on with my system. I've uninstalled and reinstalled Office to no avail.
I wanted to try out this new feature but when I did, nothing appeared to change. But then I went back to the resize and adjust screen and check this out: the preview is reflecting an inverted image as if my prior action did indeed take. Have you ever seen anything like this before?
![image](https://github.com/stevencohn/OneMore/assets/156608528/24766b40-5bf2-4e20-a9cd-56e1f69f70b2)
